### PR TITLE
Ng 34 restrict ports

### PR DESCRIPTION
**Pull request by cfboy**

### CODE REVIEW

File: GUI/window.py [hunk 1]
```diff
@@ -506,26 +505,42 @@ def save_io_ports(self, *args):
...
-                        TRAFFIC_LIGHT['port'] = num
-                    elif title == SEVEN_SEGMENT_DISPLAY['menu_title']:
-                        SEVEN_SEGMENT_DISPLAY['port'] = num
-                    elif title == ASCII_TABLE['menu_title']:
-                        if num > 4088:
-                            toast_message = 'Invalid port for ASCII Table. Valid ports [0-4088]'
+                    if is_valid_port(port):
+                        if title == TRAFFIC_LIGHT['menu_title']:
+                            update_reserved_ports(TRAFFIC_LIGHT['port'], port)
+                            TRAFFIC_LIGHT['port'] = port
```
Comment (ChristianPerez34) [inline GUI/window.py]: `TRAFFIC_LIGHT['port'] = port` <--- Do this inside of `update_reserved_ports`

And do the same for the rest of the I/O

File: constants.py [hunk 2]
```diff
@@ -128,3 +128,31 @@ def hex_to_binary(hex_num):
...
+
+]
+
+
+def is_valid_port(port):
+    """
+    Verify if port is valid
+
+    :param port: int
+    :return: Boolean
```
Comment (ChristianPerez34) [inline constants.py]: :return: bool

File: constants.py [hunk 3]
```diff
@@ -128,3 +128,31 @@ def hex_to_binary(hex_num):
...
 }
+
+RESERVED_PORTS = [
+
+]
+
+
+def is_valid_port(port):
+    """
+    Verify if port is valid
```
Comment (ChristianPerez34) [inline constants.py]: Verify if the port is available

File: constants.py [hunk 4]
```diff
@@ -128,3 +128,31 @@ def hex_to_binary(hex_num):
...
+
+    return port not in RESERVED_PORTS
+
+
+def update_reserved_ports(port_to_remove, port_to_add):
+    """
+    Update RESERVED_PORTS List.
+    
+    :param port_to_remove: 
+    :param port_to_add:
```
Comment (ChristianPerez34) [inline constants.py]:
```Python
:param port_to_remove: int
:param port_to_add: int
```

**Review by ChristianPerez34 — CHANGES_REQUESTED** [review]

File: GUI/window.py (hunk 1 above)
Comment (alejoreyes96) [reply GUI/window.py]: What do you mean? `update_reserved_ports` is used to update the stack using the old value of that I/O, not update the dictionary where `TRAFFIC_LIGHT` is located.

Comment (alejoreyes96) [reply GUI/window.py]: AH. My bad, read wrong. Read do this **instead of**, not **inside of**.

**Review by noelvalentin — APPROVED** [review]

**Review by ChristianPerez34 — APPROVED** [review]

**Review by alejoreyes96 — APPROVED** [review]

**Pull request merged**